Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:16.04 as build
+
+MAINTAINER SDF Ops Team <ops@stellar.org>
+
+RUN mkdir -p /app
+WORKDIR /app
+COPY package.json yarn.lock tsconfig.json /app/
+COPY src /app/src
+COPY public /app/public
+
+RUN apt-get update && apt-get install -y curl git make apt-transport-https && \
```

Comment:
Doing everything in 1 step ends up causing me issues every time I have to debug a production issue, cuz there are no intermediate caches—changing 1 file means reinstalling deps and rebuilding.

I much prefer if this is broken up so that files are COPY'd immediately before the step that uses them, see [stellar/laboratory](https://github.com/stellar/laboratory/pull/461/files#diff-3254677a7917c6c01f55212f86c57fbf) or [stellar/new-docs](https://github.com/stellar/new-docs/blame/master/Dockerfile) for examples. With both of these, I can e.g. change the nginx config without needing to reinstall deps or rebuild JS.